src/smt/theory_str_noodler/lia_solver.h

```diff
@@ -0,0 +1,58 @@
+#ifndef _NOODLER_LIA_SOLVER_H_
+#define _NOODLER_LIA_SOLVER_H_
+
+#include "ast/ast.h"
+#include "util/lbool.h"
+#include "smt/smt_context.h"
+
+namespace smt::noodler {
+
+    /**
+     * @brief Abstract interface for internal LIA solvers used by noodler.
+     *
+     * Implementations must provide initialization from a `context`, a method to
+     * check satisfiability of an arithmetic/length `expr` and a way to
+     * retrieve an (approximate) unsat core when available.
+     */
+    class lia_solver {
+    public:
+        /**
+         * @brief Virtual destructor.
+         */
+        virtual ~lia_solver() = default;
+
+        /**
+         * @brief Initialize the solver with formulas present in the given Z3 `context`.
+         *
+         * @param ctx The Z3 `context` from which to take asserted formulas and current assignment.
+         * @param include_ass If true, include current assignment formulas as part of initialization.
```

Comment:
Could you explain what "assignment formulas" are here? Or maybe just some hint to some other piece of code/comment. I always forget what this parameter does and this doesn't explain it properly.
Also change the name to `include_assignment` :D